dell/xps/sleep-resume/i2c-designware/default.nix

```diff
@@ -0,0 +1,45 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) mkIf mkOption types;
+  inherit (pkgs) coreutils kmod;
+
+  cfg = config.services.sleep-resume.i2c-designware;
+in {
+  options = {
+    services.sleep-resume.i2c-designware = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+           Reload the i2c_designware driver after resuming from sleep.
+         '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.sleep-resume_i2c-designware = {
```

Comment:
I would follow this pattern here https://github.com/NixOS/nixos-hardware/blob/9d87bc030a0bf3f00e953dbf095a7d8e852dab6b/dell/inspiron/5515/default.nix#L15
to avoid race conditions.

Reply:
Thanks for that, @Mic92 !
Weirdly, I realised just the other day I should've browsed the repo again to see if there was a better way.

Thanks for the pointer!

Reply:
Hi @Mic92,
I extracted the code I'd use into #526, and created #527 to update the Dell Inspiron 5515 profile to use it.
Hopefully that works for you?